openapi.yaml

```diff
@@ -0,0 +1,281 @@
+---
+swagger: "2.0"
+info:
+  description: "This api service served to scraper administrator and reader scraper services"
+  version: "1.0.0"
+  title: "Reader Scraper Admin API"
+  contact:
+    name: "Gorillab"
+    url: "https://gorillab.co/"
+    email: "hello@gorillab.co"
+host: "localhost:3001"
+basePath: "/api/v1"
+tags:
+- name: "scrapers"
+  description: "Scraper module"
+- name: "logs"
+  description: "Log module"
+schemes:
+- "http"
+- "https"
+consumes:
+- "application/json"
+produces:
+- "application/json"
+paths:
+  /scrapers:
+    get:
+      tags:
+      - "scrapers"
+      summary: "Returns all scrapers available in the database"
+      operationId: "getScrapers"
+      parameters:
+      - name: "sort"
+        in: "query"
+        description: "Sort the list of scrapers by property"
+        required: false
+        type: "string"
+        enum:
+        - "title"
+      - name: "limit"
+        in: "query"
+        description: "Limit number of scrapers return from server"
+        required: false
+        type: "integer"
+        default: 25
+        maximum: 100
+        minimum: 1
+        format: "int32"
+      - name: "page"
+        in: "query"
+        description: "How many rows to skip"
+        required: false
+        type: "integer"
+        default: 0
+        maximum: 100
+        minimum: 0
+        format: "int32"
+      - name: "query"
+        in: "query"
+        description: "Keywords to search"
+        required: false
+        type: "string"
+      responses:
+        200:
+          description: "successful operation"
+          schema:
+            type: "array"
+            items:
+              $ref: "#/definitions/Scraper"
+        default:
+          description: "error operation"
+          schema:
+            $ref: "#/definitions/Error"
+      security:
+            - AdminSecurity: []
+  /scrapers/register:
+    post:
+      tags:
+      - "scrapers"
+      summary: "Register a new scraper"
+      operationId: "register"
+      parameters:
+      - in: "body"
+        name: "body"
+        required: true
+        schema:
+          $ref: "#/definitions/Scraper"
+      responses:
+        200:
+          description: "successful operation"
+          schema:
+            $ref: "#/definitions/Scraper"
+        default:
+          description: "error operation"
+          schema:
+            $ref: "#/definitions/Error"
+      security:
+        - api_key: []
+  /scrapers/{id}/createPosts:
```

Comment:
the only info for authentication is the scraper id. put it in the header `reader-scraper-id`. the admin will check if it exists, then proceed to import uploaded data. otherwise response 401 unauthorized